pfrl/utils/batch_states.py

```diff
@@ -30,4 +30,7 @@ def batch_states(
     """
     features = [phi(s) for s in states]
     # return concat_examples(features, device=device)
-    return _to_recursive(default_collate(features), device)
+    collated_features = default_collate(features)
+    if isinstance(collated_features, list):
```

Comment:
```suggestion
    if isinstance(features[0], tuple):
```
I think collated_features should be casted to tuple only when `isinstance(features[0], tuple)`, so that `batch_states` keeps the original sequence type as much as possible.